### DISCUSSION TITLE
使用VScode+texlive2021在命令行使用latexmk -xelatex -outdir=out zjuthesis 仍无法显示参考文献#23的问题

### DISCUSSION THREAD

**Discussion by Luorifeihua** [TheNetAdmin/zjuthesis#370] [Tool Usage]
我在使用VScode按readme和FAQ以及issue#23帖子中的内容尝试后仍无法解决参考文献无法显示的问题。
  经测试，在config\packages.tex文件下找到\usepackage[...]{biblatex}中在原有的选项前加入[backend=bibtex]，再次运行即可显示参考文献。问题出在lateworkShop在latexmk命令执行时使用bibtex对.bib文件进行处理，而biblatex宏包默认参考文献后端为biber。
  但是，查阅biblatex手册后，对backend=bibtex的修改可能会导致一些限制（没能理解），故而项问问是否能将latexmk编译器的默认后端改为biber以匹配biblatex宏包的默认后端。
<img width="998" alt="image" src="https://github.com/TheNetAdmin/zjuthesis/assets/69577471/5b4336d8-3b2d-4a89-96f7-422b562ce297">

**Comment by Chenyanlu99** [comment]
同Vscode，实测`xelatex->biber->xelatex*2`可以编译成功

**Comment by YorkChin02** [comment]
限制就是，参考文献无 et al. 
我是Mac vscode，[backend=biber] 就会失败且不报错，[backend=bibtex] 成功但参考文献无 et al. 
用Windos vscode，[backend=biber] 、[backend=bibtex] 均成功，[backend=bibtex] 时参考文献无 et al.

**Reply by YorkChin02** [reply]
我在zjuthesis.tex根目录下用终端运行命令行“latexmk -xelatex -outdir=out zjuthesis”可以在Mac上成功编译出有et al.的参考文献。
然后就可以正常使用教程中的latexmkrc的编译配方了。可能是运行命令行清除了一些旧的文件。
